### PR TITLE
Don't render retrieval action for title generation.

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

Don't render retrieval action for title generation.

Context [here](https://github.com/dust-tt/tasks/issues/717#issuecomment-2084529152) and [here](https://dust4ai.slack.com/archives/C050SM8NSPK/p1718016057015179).

### CODE REVIEW

**Review by spolu — CHANGES_REQUESTED** [review]
Can we make it more general and exclude all actions 🙏

**Comment by lasryaric** [comment]
>Can we make it more general and exclude all actions 🙏

Yes I hesitated so went for the simple version before getting your feedback.

File: front/lib/api/assistant/generation.ts [hunk 1]
```diff
@@ -91,7 +93,9 @@ export async function renderConversationForModel({
           content: m.content,
         });
       }
-      if (m.action) {
+      // There are contexts where we want to exclude the actions from the rendering.
+      // Eg: During the conversation title generation step.
+      if (m.action && !excludeActions) {
```
Comment (spolu) [inline front/lib/api/assistant/generation.ts:98]: If this is an action and excludeActions is true we will fall in the else if all the way to the assert never no?

Comment (spolu) [reply front/lib/api/assistant/generation.ts:98]: I think the excludeAction condition must be inside the that if block?

Comment (spolu) [reply front/lib/api/assistant/generation.ts:98]: Nevermind me I was one if block off

**Review by spolu — APPROVED** [review]
LGTM

**Pull request merged**